Sources/StreamChat/Workers/MessageUpdater.swift

```diff
@@ -255,15 +255,48 @@ class MessageUpdater: Worker {
         messageId: MessageId,
         completion: ((Error?) -> Void)? = nil
     ) {
+        var reaction: MessageReactionDTO?
+        let version = UUID().uuidString
+
         let endpoint: Endpoint<EmptyResponse> = .addReaction(
             type,
             score: score,
             enforceUnique: enforceUnique,
             extraData: extraData,
             messageId: messageId
         )
-        apiClient.request(endpoint: endpoint) {
-            completion?($0.error)
+
+        database.write { session in
+            do {
+                reaction = try session.addReaction(
+                    to: messageId,
+                    type: type,
+                    score: score,
+                    extraData: extraData
+                )
+                completion?(nil)
+            } catch {
+                log.warning("Failed to add the reaction to the database: \(error)")
+                completion?(error)
+            }
+
+            if let reaction = reaction {
+                reaction.localState = .sending
+                reaction.version = version
+            }
+        } completion: { _ in
```

Comment:
```suggestion
        } completion: { error in
```

We need to check the error in the completion block. This is what will make sure it was successfully written to the DB or not.